lib/kennel/filter.rb

```diff
@@ -2,32 +2,45 @@
 
 module Kennel
   class Filter
+    attr_reader :project_filter, :tracking_id_filter
+
     def initialize
-      project_filter
-      tracking_id_filter
+      # build early so we fail fast on invalid user input
+      @tracking_id_filter = build_tracking_id_filter
+      @project_filter = build_project_filter
+    end
+
+    def filter_projects(projects)
+      filter_resources!(projects.dup, :kennel_id, project_filter, "projects", "PROJECT")
```

Comment:
Let's keep the dup-ness as an implementation detail of the private method

```suggestion
      filter_resources(projects, :kennel_id, project_filter, "projects", "PROJECT")
```